### ISSUE TITLE
Expression Consistency

### ISSUE THREAD

**Issue by JohnCoene** [blockr-org/blockr#87]
I think expressions should be more consistent, this would be clearer and allow simplifying the code.

For instance the select block's expression looks like:

https://github.com/blockr-org/blockr/blob/9c88b24a7597ff86af5b69714f25952778d391da/R/block.R#L328

But the plot block looks like:

https://github.com/blockr-org/blockr/blob/9c88b24a7597ff86af5b69714f25952778d391da/R/block.R#L657

This leads to two different eval methods:

One to which we pass the `data` (plot block) but one to which we don't (select block).

I would make this more consistent, I don't understand why we don't pass the data to any transform block, I would imagine someone might create a block that requires it.

**Comment by nbenn** [comment]
@JohnCoene I agree. The quoted expression for a block should always be a function call where the result of the previous block can be piped as (first?) argument.

**Comment by JohnCoene** [comment]
To me it'd be better to pass `data` but at the same time I was thinking about your comment on #81 and I guess the generated code would not be as neat (for a whole stack).

Currently we can generate something like `data %>% select(col) %>% filter(...)` if the expression for the select block becomes `select(data, col)` the generated stack code won't be as neat?
